project1/application.py

```diff
@@ -21,64 +21,110 @@
 app.config["SESSION_TYPE"] = "filesystem"
 Session(app)
 
-
 @app.route("/")
 def index():
-    if 'name' not in session :
-        return redirect(url_for('register'))
-    elif session['name'] :
-        return render_template("home.html")
-    
+	if 'name' not in session :
+		return redirect(url_for('register'))
+	elif session['name'] :
+		xyz = Book.query.all()
+		return render_template("home.html", xyz=xyz, flag1=1)
+
+@app.route("/search", methods = ["POST"])
+def search():
+		if request.method == 'POST':
+			key = request.form.get('search2')
+			search_list = []
+			filtered_list1 = Book.query.get(key)
+			if filtered_list1:
+				search_list.append(2)
+				return render_template("home.html", filtered_list=filtered_list1, flag2=1)
+			# else:
+			# 	return render_template("home.html", filtered_list=filtered_list1, flag4=1)
```

Comment:
Does search return partially matched strings? Also, can you think of simplifying the logic.